daemon/containerd/service.go

```diff
@@ -37,14 +39,24 @@ type RegistryConfigProvider interface {
 	IsInsecureRegistry(host string) bool
 }
 
+type ImageServiceConfig struct {
+	Client        *containerd.Client
+	Containers    container.Store
+	Snapshotter   string
+	HostsProvider RegistryHostsProvider
+	Registry      RegistryConfigProvider
+	EventsService *daemonevents.Events
+}
+
 // NewService creates a new ImageService.
-func NewService(c *containerd.Client, containers container.Store, snapshotter string, hostsProvider RegistryHostsProvider, registry RegistryConfigProvider) *ImageService {
+func NewService(config ImageServiceConfig) *ImageService {
 	return &ImageService{
-		client:          c,
-		containers:      containers,
-		snapshotter:     snapshotter,
-		registryHosts:   hostsProvider,
-		registryService: registry,
+		client:          config.Client,
+		containers:      config.Containers,
+		snapshotter:     config.Snapshotter,
+		registryHosts:   config.HostsProvider,
+		registryService: config.Registry,
+		eventsService:   config.EventsService,
 	}
 }
```

Comment:
Please tell me there's some other PR which does _something_ in the `NewService` constructor aside from copying all the fields from one struct into corresponding fields on another struct which only differs in exportedness of the fields.

Reply:
Good call; looking at the fork.. I don't think there is; https://github.com/rumpl/moby/blob/5d0d4c2cb2fef08f72ac929335d0f3918943c05e/daemon/containerd/service.go#L25-L49

So "plus one" on just using a struct literal and removing the `ImageServiceConfig`

Reply:
Maybe the idea here was to not export the `client`, `container` etc?

Reply:
The function was starting to get too many parameters and I don't want these fields to be public